src/Components/Sort.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import {
+  Sort as SortType,
+  SortState,
+  buildSort,
+  buildRelevanceSortCriterion,
+  buildDateSortCriterion,
+  SortOrder,
+  SortByRelevancy,
+  SortByDate
+} from "@coveo/headless";
+import { headlessEngine } from "../Engine";
+import FormControl from "@material-ui/core/FormControl";
+import NativeSelect from "@material-ui/core/NativeSelect";
+
+export default class Sort extends React.Component {
+  private headlessSort: SortType;
+  state: SortState;
+  relevanceSortCriterion: SortByRelevancy = buildRelevanceSortCriterion();
+  dateDescendingSortCriterion: SortByDate = buildDateSortCriterion(
+    SortOrder.Descending
+  );
+  dateAscendingSortCriterion: SortByDate = buildDateSortCriterion(
+    SortOrder.Ascending
+  );
+
+  constructor(props: any) {
+    super(props);
+
+    this.headlessSort = buildSort(headlessEngine, {
+      initialState: {
+        criterion: this.relevanceSortCriterion
+      }
+    });
+
+    this.state = this.headlessSort.state;
+  }
+
+  componentDidMount() {
+    this.headlessSort.subscribe(() => this.updateState());
+  }
+
+  updateState() {
+    this.setState(this.headlessSort.state);
+  }
+
+  render() {
+    return (
+      <FormControl>
+        <NativeSelect
+          onChange={(event: any) => {
+            switch (event.target.value) {
+              case "relevance":
+                this.headlessSort.sortBy(this.relevanceSortCriterion);
+                break;
+              case "datedescending":
+                this.headlessSort.sortBy(this.dateDescendingSortCriterion);
+                break;
+              default:
+                this.headlessSort.sortBy(this.dateAscendingSortCriterion);
+                break;
+            }
+          }}
```

Comment:
For some reason, making this a separate handleChange function breaks, because of something to do with mounting/unmounting. The only way to make this work was to use a lambda function

Reply:
It should be possible to extract the logic into a function and just use a lambda to call it:

```
render() {
    <NativeSelect onChange={(event: any) => this.updateSortCriterion(event)}
}
```

Reply:
So I looked into this and it was a scoping issue. When you use `this` within the context of a method, it will be interpreted as the method, not the class object. We can get around this by binding the class object to the method when we call it:

`<NativeSelect onChange={this.handleChange.bind(this)}>`